### PR TITLE
port just restart fix

### PR DESCRIPTION
**Pull request by evanmcc**
just the major fix from #289, without all the debugging and new test.

### CODE REVIEW

File: src/group/libp2p_group_relcast_server.erl [hunk 1]
```diff
@@ -357,7 +357,7 @@ handle_info(force_close, State=#state{}) ->
     %% down the group by exiting the supervisor.
     spawn(fun() ->
                   lager:info("removing group for force_close timeout"),
-                  libp2p_swarm:remove_group(State#state.tid, State#state.group_id)
+                  libp2p_group_mgr:remove_group(State#state.tid, State#state.group_id)
```
Comment (Vagabond) [inline src/group/libp2p_group_relcast_server.erl:360]: Why is this needed, given that libp2p_swarm:remove_group calls libp2p_group_mgr:remove_group?

Comment (Vagabond) [reply src/group/libp2p_group_relcast_server.erl:360]: Actually, this calls via the TID, but libp2p_group_mgr:remove_group expects a pid as the first argument?

File: src/group/libp2p_group_relcast_server.erl [hunk 2]
```diff
@@ -357,7 +357,8 @@ handle_info(force_close, State=#state{}) ->
     %% down the group by exiting the supervisor.
     spawn(fun() ->
                   lager:info("removing group for force_close timeout"),
-                  libp2p_swarm:remove_group(State#state.tid, State#state.group_id)
+                  Mgr = libp2p_group_mgr:remove_group(State#state.tid),
```
Comment (Vagabond) [inline src/group/libp2p_group_relcast_server.erl:360]: D you mean `libp2p_group_mgr:mgr` here?

Comment (evanmcc) [reply src/group/libp2p_group_relcast_server.erl:360]: lol.  So I reverted the change here totally, and it broke again as expected.  trying with the fix you suggested, I suspect that it will also break.

UPDATE: before I finished typing the above sentence, the "corrected" code caused the same crash.

Comment (Vagabond) [reply src/group/libp2p_group_relcast_server.erl:360]: I think it worked because it was failing to close.
